### ISSUE TITLE
Improve Footer of Privly applications website

### ISSUE THREAD

**Issue by opg7371** [privly/privly-applications#318]
The footer is too simple.It should be updated .

**Comment by opg7371** [comment]
![new_footer](https://cloud.githubusercontent.com/assets/10517011/12612767/295dce6a-c51b-11e5-8715-5334a84e16b8.jpg)
Please give me the review regarding the footer and any else change that the organization wants to see.

**Comment by smcgregor** [comment]
* The copyright needs updating to 2016
* This is using a lot of vertical space without using any of the horizontal space. If you use bootstrap's column system the footer should stack automatically when the width is too small.

**Comment by opg7371** [comment]
copyright is updated.
The footer in normal cases look like footer.png
![footer](https://cloud.githubusercontent.com/assets/10517011/12640202/c666f19c-c5ce-11e5-93f1-ac7d30835ed3.png)


The footer when width is too small is footer_min.png
![footer_min](https://cloud.githubusercontent.com/assets/10517011/12640210/cde46b2a-c5ce-11e5-851a-97591aec65a0.png)

**Comment by smcgregor** [comment]
It looks like you aren't using Bootstrap's column system? The column content should start at the same vertical position.

**Comment by opg7371** [comment]
I am using the Bootstrap's column system. Please see the screenshot attached.
For normal view - 
![Uploading footer_col_privly.png…]![footer_col_privly](https://cloud.githubusercontent.com/assets/10517011/12697789/b8a1ee38-c7b1-11e5-96a3-1deccff6c102.png)

When the  width is too small then it looks like..
()
()
![footer_min_privly](https://cloud.githubusercontent.com/assets/10517011/12697785/9cd72934-c7b1-11e5-9916-0bf075a7bfa0
0bf075a7bfa0.png)

**Comment by smcgregor** [comment]
Can you post a snippet of the HTML used to produce this column layout? It looks like everything is in a single page-width column, `col-xs-12`, but the different sections have center, right, and left alignment (from top to bottom).

**Comment by opg7371** [comment]
This is the code for footer.

<footer> 
 <div id="footer">
   <div class="container text-center">
    <div class="row">
     <br/>
     <div class="text-center">You are viewing a Privly
     <span style="vertical-align: super;font-size: xx-small">
      ®
     </span>
     Application
     </div>
     </div>
     <br/>
     <div class="row">
     <div class="col-sm-3">
     </div>
       <div class=" col-xs-12 col-sm-3" style="padding-top:5px">
         <button>
       <a href="https://www.priv.ly/pages/donate.html">
        <span class="glyphicon glyphicon-gift">
        </span>
        Donate
       </a>
       <br/>
       to build a more private web.
      </button>
       </div>
       <div class=" col-xs-12 col-sm-3" style="padding-top:5px">
         <button>
       <a href="https://priv.ly/pages/about" target="_blank">
        Learn More.
       </a>
       <br/>
       Learn more about Privly here.
      </button>
       </div>
       <div class="col-sm-3">
       	<div class="row">
   <div class="col-xs-12" style="font-size:45px margin-top:70px"> Follow us on:</font-size></div>
    	</div>
<ul class="list-inline">
      <li class="icon-soc">
       <a href="https://twitter.com/privly">
        <img alt="Twitter" src="../shared/images/soc/twitter.png"/>
       </a>
      </li>
      <li class="icon-soc">
       <a href="https://www.facebook.com/shareprivately/timeline">
        <img alt="FaceBook" src="../shared/images/soc/facebook.png"/>
       </a>
      </li>
      <li class="icon-soc">
       <a href="https://groups.google.com/forum/#!forum/privly">
        <img alt="Google" src="../shared/images/soc/google.png"/>
       </a>
      </li>
     </ul>
     </div>
     </div>
     <p class="text-left" style="padding-top:40px">
      <a class="legal_nav" href="http://priv.ly" target="_blank">
       <strong>
        Terms of Service
       </strong>
      </a>
      |
      <a class="legal_nav" href="http://priv.ly" target="_blank">
       DMCA
      </a>
      | © Copyright 2016.
     </p>
       </div>
   </div>
  </div>
</footer>

**Comment by smcgregor** [comment]
* Don't use tab characters
* Make sure any HTML you share is properly nested and indented
* It is very difficult to give feedback when it isn't formatted already
* I recommend using Pastebin (or something similar) to share code that is formatted and syntax highlighted
* Make sure you only have 3 columns within 1 row inside a container

Regarding the build.py file, that is a different PR. You should comment there. I haven't been able to test it because I need to debug why `pip3 install -r requirements...` is not working on my environment.

**Comment by opg7371** [comment]
![privly_footer_new](https://cloud.githubusercontent.com/assets/10517011/12713283/b3581ebe-c8f3-11e5-8b3f-db649686c2af.png)
![privly-footer-new-min](https://cloud.githubusercontent.com/assets/10517011/12713290/b9fe457c-c8f3-11e5-901d-1468f4a7403d.png)

This is the new privly footer i designed.I took special care for keeping only 3 columns within a row.
But inorder to keep the 2 buttons(donate and learn more) one below another when the width is too small,so i used nested columns inside of the row .The code for the following is http://pastebin.com/EQwHEP34
It is properly organized.

I am also looking for some changes.I want to add a github repo image and a main website image(which link to the main website) too at the left inorder to provide symmetry to our website.

looking forward to our suggestions.
Please tell me if there is to add anything else.

**Comment by smcgregor** [comment]
* There are still indentation issues on the Pastebin link, which makes it difficult to follow.
* When you are evaluating your designs, pretend like there is a box surrounding each of the elements formed by infinitely long lines. Now think of how you can reduce the number of these lines that are visible. This is not a hard and fast requirement, but it is useful for thinking about alignment of elements.

**Comment by opg7371** [comment]
I have removed the indentation issues. The link for the same is http://pastebin.com/pVGBPcjU

**Comment by smcgregor** [comment]
* Why the increasing size of corporate logos from left to right?
* The buttons would likely look a bit better if they had the same dimensions. You could also add an icon for the "learn more" button.
* I think the "Terms of Service" section is not using Bootstrap's column layout?

**Comment by opg7371** [comment]
The increasing size of corporate logos from left to right was something i thought would look good.Now the dimensions of corporate logos are same.
![footer_min_neww](https://cloud.githubusercontent.com/assets/10517011/12776208/17b721ca-ca7a-11e5-8a95-fa85940f2a6a.png)
![footer_neww](https://cloud.githubusercontent.com/assets/10517011/12776209/17b8f41e-ca7a-11e5-9636-c51d27c75081.png)

I think that these 2 buttons should be kept inorder to maintain Symmetry.The footer will not look too good with only one button at the center.But if you insist then i will remove the learn more button and will add a image icon in place of it.

Yes you are right,The "Terms of Service " section was not using the Bootstrap column layout system.But now it is following it.The link for the same - http://pastebin.com/ZiHPqiK3

Let me know about the Learn more button

**Comment by smcgregor** [comment]
* By "adding an icon" I did not mean replacing the button. The "Donate" button has a present in it. You could add an icon to the learn more button.
* The equal sized buttons does look better.
* The alignment under the full-width layout does not look good. In particular, it looks like a mistake to have the terms in their current position. It might look better if you combine them with the "You are viewing..." text somehow.

**Comment by opg7371** [comment]
I will add the icon to the button learn more.

Can you be more specified regarding the third point and elaboratee the "you are viewing .." part since it is present beforehand.I am not getting of how to combine them.

**Comment by smcgregor** [comment]
You can reorder, rearrange, or even cut something entirely from the footer if you think it is best. Two options are moving it into a column, or changing it to a two column layout and moving the "Terms" text out of the column layout.

**Comment by opg7371** [comment]
sir can we chat at irc if you are free not regarding this issue.

**Comment by opg7371** [comment]
Sir,pardon me but i am not sure what 'moving the "Terms" text out of the column layout ' means but i guess you want me to move the terms and other section to the complete left.
I have done something like this
![footer_new_left](https://cloud.githubusercontent.com/assets/10517011/12856217/cbc861b0-cc69-11e5-9a02-bcffd94e7678.png)
![footer_feb-06-min](https://cloud.githubusercontent.com/assets/10517011/12856226/d4dcbfe4-cc69-11e5-9762-91613357be0e.png)

Since the text part was already in the 2 column layout , hence i was not sure about the moving .. part.
Please clarify me on this.
Is this layout okay?

**Comment by smcgregor** [comment]
My suggestion was to combine or eliminate some of the text. You could then shift to a 2 column layout with another row above or below the 2 column layout containing the text. The layout of the terms of service links currently looks awkward.

**Comment by opg7371** [comment]
sir in that case,we can change the terms and service to terms or TOS or terms and condition
However my suggestion is that we can shift the terms and other part to the center and we can include your organization's address in left part which will give it a better look.what do you say?

**Comment by smcgregor** [comment]
How about instead of a mailing address (which is not useful to most people), we include another link to the [safety page](https://priv.ly/pages/safety)?

**Comment by opg7371** [comment]
![footer_2016-feb-09-min](https://cloud.githubusercontent.com/assets/10517011/12917417/4ba34cd6-cf5f-11e5-89d7-82ba65635ed0.png)
![footer_feb_9_2016](https://cloud.githubusercontent.com/assets/10517011/12917422/54c4f1ac-cf5f-11e5-876f-edc75c77ef1d.png)

I have uploaded the link to safety page and shifted the entire terms part to the center

looking forward to your suggestion :)

**Comment by smcgregor** [comment]
The "Priv.ly" logo links to the safety page? I think the logo looks good there, but it is not obvious where clicking it takes you.

**Comment by opg7371** [comment]
yes it do links to the safety page...how about we add a link by adding it near the term's part
Is that okay?

**Comment by smcgregor** [comment]
Near the terms or add another full sized button. It is important enough to be prominently displayed.

**Comment by opg7371** [comment]
So we can replace the Learn More Button with the safety button and add learn more near the terms part.
or we can give it a special effect near the term's part.In my view 3 buttons will look totally odd.
What do you say?

**Comment by smcgregor** [comment]
You have likely tried more configurations of the buttons than I have, so I'll defer to your choice.

**Comment by opg7371** [comment]
I want your feedback on this one whenever you are free.
![footer-safety-inclucded-12-2012](https://cloud.githubusercontent.com/assets/10517011/13001680/0f6c9c14-d18e-11e5-9646-85f8d2d0dec5.png)
![footer-safety-included-12-2016](https://cloud.githubusercontent.com/assets/10517011/13001681/0f7101fa-d18e-11e5-9e1b-e6e371f6a84d.png)

Since it is one of the key component of footer,hence i prefer to keep it in a button
what do you say.

**Comment by opg7371** [comment]
some more views on which i want your suggestion
![footer-include-safety -3](https://cloud.githubusercontent.com/assets/10517011/13001752/de10ebd8-d18e-11e5-843c-a917d01a69d2.png)
![footer-include-safety-3-12-2016](https://cloud.githubusercontent.com/assets/10517011/13001753/de356fc6-d18e-11e5-8321-397a41f3c90b.png)

**Comment by smcgregor** [comment]
I think the three buttons in a row look best. What are your thoughts?

**Comment by opg7371** [comment]
yes...so should i submit the code....or any other modification is also required

**Comment by smcgregor** [comment]
You still need to make some changes. Take another pass at it and refine your chosen design.

**Comment by opg7371** [comment]
![privly-footer-feb-18](https://cloud.githubusercontent.com/assets/10517011/13185970/cd551432-d769-11e5-88fc-fa3916a514aa.png)

This is the footer i created just now.have icons aligned properly
Waiting for the review...:)
Also i have the polished code for this

**Comment by smcgregor** [comment]
Do the buttons have equal width? It matters when viewed on the phone.

**Comment by opg7371** [comment]
Yes they are of equal size
![footer-min-feb-19](https://cloud.githubusercontent.com/assets/10517011/13195780/9d00fd92-d7e0-11e5-9b46-c04d7a2ce87a.png)
Now what is your review....??

**Comment by smcgregor** [comment]
They are a few pixels off of being the same width. I will be tweaking the text in them, so they will need to have a constant width anyway.

It looks like the color on the logo is not quite matching?

**Comment by opg7371** [comment]
![footer_min_feb_25](https://cloud.githubusercontent.com/assets/10517011/13318123/e31938a6-dbde-11e5-9d83-b0738aff5999.png)

![privly-footer-new-max](https://cloud.githubusercontent.com/assets/10517011/13317959/0bc04a66-dbde-11e5-8e38-32b0e923d638.png)

I thought i posted them 2 days before....but anyways i want your view on this...
changed the icon to lobster view

**Comment by prastut** [comment]
@smcgregor  @opg7371 can we make the footer a little simpler like this.

![fireshot capture 11 - houseafic - https___www houseafic com_](https://cloud.githubusercontent.com/assets/10279686/13334361/9e799184-dc31-11e5-9254-b82dcc670e11.png)
- Do away with sharing icons. 
- Make the logo prominent and add copyrights below it. 
- Rest of the things as shown by the image.

**Comment by smcgregor** [comment]
The social network links are not for sharing, I plan on linking them to Privly's pages on those sites.

I like @prastut's solution more for the web-hosted version of the application, but I like @opg7371's solution more for the browser extension.

@irdan @vshivam @spectralsun Any thoughts?

**Comment by vshivam** [comment]
For me, the existing footer is fine. It's clean and gets the point across without being ostentatious. Unless the priority is to add social icons I don't see a reason for updating it. 

From the mobile app's perspective, I think there's too much going on in @opg7371 's implementation. It takes half the height of the viewport!

**Comment by prastut** [comment]
@opg7371 I recommended using Bootstrap's predefined classes for nifty CSS hacks, as well as not writing inline styles. Noticed them in the footer.txt file you provided.

**Comment by opg7371** [comment]
@prastut they were just were for model...ie how the footer will look....regarding once the look of the footer will be accepted...the code will be done accordingly.

**Comment by opg7371** [comment]
@smcgregor should i open a pull request?

**Comment by smcgregor** [comment]
Good points, @vshivam.

I think I could make a case for three different layouts on the footer. Mobile needs something simple (what we have) because anything else takes up too much of the viewport. The web-hosted version needs links to carefully selected resources to explain what is happening. The extension version may need to have a different set of links.

@opg7371I think your design is somewhere between a design for the web-hosted version and a design for the extension version. Which case do you want to solve?

**Comment by opg7371** [comment]
@smcgregor I would like to solve the design for extension version first
